include/m_common_data.h

```diff
@@ -159,86 +150,90 @@ typedef struct FamicomEmuCommonData {
     /* 0x0EF6C */ Mail unk_0EF6C[5];
     /* 0x0F2A0 */ UNK_TYPE1 unk_0F2A0[0x17C];
     /* 0x0F41C */ SnowmanData snowmanData[SNOWMAN_SAVE_COUNT];
     /* 0x0F428 */ UNK_TYPE1 unk_F428[0x10];
-    /* 0x0F438 */ u8 station_type; /* train station type */
-    /* 0x0F439 */ u8 unk0F439[0x3];
-    /* 0x0F43C */ u16 deposit[FG_BLOCK_X_NUM * FG_BLOCK_Z_NUM][UT_Z_NUM]; /* flags for which items are buried around town */
+    /* 0x0F438 */ u8 stationType;
+    /* 0x0F439 */ u8 unk_F439[0x3];
+    /* 0x0F43C */ u16 deposit[FG_BLOCK_X_NUM * FG_BLOCK_Z_NUM][UT_Z_NUM]; // flags for which items are buried around town
     /* 0x0F7FC */ lbRTC_time_c unk_0F7FC;
-    /* 0x0F804 */ mPr_mother_mail_info_c mother_mail[PLAYER_NUM];
-    /* 0x0F83C */ u8 unk0F83C[0x8];
-    /* 0x0F844 */ FamicomEmuCommonData famicom_emu_common_data;
-    /* 0x0F86A */ u8 unk0F86A[0x32];
+    /* 0x0F804 */ MotherMailInfo motherMailInfo[PLAYER_NUM];
+    /* 0x0F83C */ u8 unk_0F83C[0x8];
+    /* 0x0F844 */ FamicomEmuCommonData famicomEmuCommonData;
+    /* 0x0F86A */ u8 unk_0F86A[0x32];
     /* 0x0F89C */ lbRTC_time_c unk_0F89C;
     /* 0x0F8A4 */ lbRTC_time_c unk_0F8A4;
-    /* 0x0F8AC */ UNK_TYPE1 unk0F8AC;
-    /* 0x0F8AD */ u8 snowmanYear; // Year last snowman was built.
+    /* 0x0F8AC */ UNK_TYPE1 unk_0F8AC;
+    /* 0x0F8AD */ u8 snowmanYear;  // Year last snowman was built.
     /* 0x0F8AE */ u8 snowmanMonth; // Month last snowman was built.
-    /* 0x0F8AF */ u8 snowmanDay; // Day last snowman was built.
-    /* 0x0F8B0 */ u8 snowmanHour; // Hour last snowman was built.
-    /* 0x0F8B1 */ UNK_TYPE1 unk0F8B1[0x74F];
+    /* 0x0F8AF */ u8 snowmanDay;   // Day last snowman was built.
+    /* 0x0F8B0 */ u8 snowmanHour;  // Hour last snowman was built.
+    /* 0x0F8B1 */ UNK_TYPE1 unk_0F8B1[0x74F];
     /* 0x10000 */ u8 unk_10000; // named "game_started" in AC GCN decomp
     /* 0x10001 */ u8 unk_10001;
-    /* 0x10002 */ u8 unk10002[0x1];
-    /* 0x10003 */ u8 player_no;
+    /* 0x10002 */ u8 unk_10002[0x1];
+    /* 0x10003 */ u8 playerNumber;
     /* 0x10004 */ s32 unk_10004; // named "last_scene_no" in AC GCN decomp
     /* 0x10008 */ UNK_TYPE1 unk_10008[0x44];
-    /* 0x1004C */ CommonData_unk_1004C *unk_1004C;
+    /* 0x1004C */ CommonData_unk_1004C* unk_1004C;
     /* 0x10050 */ UNK_TYPE1 unk_10050[0x28];
-    /* 0x10078 */ CommonData_unk_10078 *unk_10078;
+    /* 0x10078 */ CommonData_unk_10078* unk_10078;
     /* 0x1007C */ UNK_TYPE1 unk_1007C[0x1C];
-    /* 0x10098 */ CommonData_unk_10098 *unk_10098;
+    /* 0x10098 */ CommonData_unk_10098* unk_10098;
     /* 0x1009C */ UNK_TYPE1 unk_1009C[0x4];
     /* 0x100A0 */ struct ToolClip* toolClip;
     /* 0x100A4 */ UNK_TYPE1 unk_100A4[0x40];
-    /* 0x100E4 */ CommonData_100E4_Func* unk_100E4;
-    /* 0x100E8 */ u8 unk100E8[0x24];
-    /* 0x1010C */ Time_c time;
-    /* 0x10138 */ Private_c* now_private;
-    /* 0x1013C */ u8 unk1013C[0x4];
+    /* 0x100E4 */ CommonData_unk_100E4* unk_100E4;
```

Comment:
```suggestion
    /* 0x100E4 */ CommonData_100E4_Func unk_100E4;
```

I know this is merged now but feels like the place to mention it, I think the issue here was that this shouldnt be a pointer to a pointer right?

Reply:
or is there any reason it should be a struct i guess

Reply:
animal crossing has a unique system for trying to template out parts of the code. These structs are called "clips", which store callback function pointers for other files to use. There are other clips for structures, tools, or npcs, though I'm not entirely sure about how they work so I've been leaving them with the unk_xxx names